CMakeLists.txt

```diff
@@ -62,7 +62,7 @@ set(IREE_TRACING_MODE ${IREE_TRACING_MODE_DEFAULT} CACHE STRING "Tracing feature
 
 # TODO(#8469): remove the dependency on cpuinfo entirely.
 set(IREE_ENABLE_CPUINFO_DEFAULT ON)
-if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+if(CMAKE_SYSTEM_NAME MATCHES "Darwin|Emscripten|Windows|WindowsStore")
   set(IREE_ENABLE_CPUINFO_DEFAULT OFF)
```

Comment:
Thanks for this! We can clean up uses of `IREE_ENABLE_CPUINFO` separately from this PR (or you can include here... that would involve touching more files)

For example: https://github.com/openxla/iree/blob/7e06b6e5ece2adcda91352331db817adaa058e85/build_tools/cmake/build_runtime_emscripten.sh#L41

Reply:
I'll get the emscripten ones too

Reply:
there's too many python/build scripts and they override the flag - I don't like that, and leave it to someone who uses those scripts to figure out what to do about it until cpuinfo is gone for good. at this point it should only be used on linux-like systems.